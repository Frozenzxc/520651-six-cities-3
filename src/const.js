const OfferType = {
  ALL: `cities`,
  FAVORITES: `favorites`,
  NEARBY: `near`,
};

const MAX_NEARBY_OFFERS = 3;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const ReviewPostingStatus = {
  POSTED: `POSTED`,
  ERROR: `ERROR`,
};

const AppRoute = {
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
  PROPERTY: `/offer/:id`,
  ROOT: `/`,
};

export {MAX_NEARBY_OFFERS, AuthorizationStatus, AppRoute, ReviewPostingStatus, OfferType};
