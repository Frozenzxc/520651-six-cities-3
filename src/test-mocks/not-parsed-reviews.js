/* eslint-disable camelcase */
export default [
  {
    id: 1,
    user: {
      id: 17,
      is_pro: false,
      name: `Emely`,
      avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`
    },
    rating: 4,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-02-23T08:36:26.433Z`
  },
  {
    id: 2,
    user: {
      id: 14,
      is_pro: true,
      name: `Corey`,
      avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg`
    },
    rating: 3,
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-02-23T08:36:26.433Z`
  }
];
