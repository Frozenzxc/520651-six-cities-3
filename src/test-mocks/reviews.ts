import {Review} from "../types";

export const reviews: Review[] = [
  {
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-02-23T08:36:26.433Z`,
    id: 1,
    rating: 4,
    user: {
      avatar: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`,
      id: 17,
      isPro: false,
      name: `Emely`
    }
  },
  {
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-02-23T08:36:26.433Z`,
    id: 2,
    rating: 3,
    user: {
      avatar: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg`,
      id: 14,
      isPro: true,
      name: `Corey`
    }
  }
];
