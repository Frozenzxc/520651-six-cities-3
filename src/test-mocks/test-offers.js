export default [
  {
    bedroomsCount: 4,
    coordinates: [52.3909553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `John`,
      super: false,
    },
    id: 1,
    maxGuests: 6,
    options: [`Wifi`, `Washing machine`, `Towels`, `Heating`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    premium: false,
    price: 200,
    rating: 3,
    reviews: [
      {
        date: `2020-02-22T00:00:00.000Z`,
        reviewID: 140,
        name: `Max`,
        rating: 1,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-24T00:00:00.000Z`,
        reviewID: 151,
        name: `John`,
        rating: 4,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 160,
        name: `Victoria`,
        rating: 3,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-03-02T00:00:00.000Z`,
        reviewID: 145,
        name: `Bob`,
        rating: 5,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    type: `house`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.369553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: false,
    },
    id: 2,
    maxGuests: 4,
    options: [`Wifi`, `Towels`, `Heating`, `Cabel TV`, `Fridge`],
    premium: true,
    price: 300,
    rating: 1,
    reviews: [
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 156,
        name: `Alex`,
        rating: 4,
        src: `img/avatar-angelina.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-24T00:00:00.000Z`,
        reviewID: 198,
        name: `Dana`,
        rating: 1,
        src: `img/avatar-angelina.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Wood and stone place`,
    type: `hotel`,
  },
  {
    bedroomsCount: 1,
    coordinates: [52.3909553943508, 4.929309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Bob`,
      super: true,
    },
    id: 3,
    maxGuests: 2,
    options: [`Wifi`, `Towels`, `Cabel TV`],
    premium: false,
    price: 450,
    rating: 3,
    reviews: [
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 176,
        name: `Max`,
        rating: 2,
        src: `img/avatar-angelina.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      super: false,
    },
    id: 4,
    maxGuests: 5,
    options: [`Wifi`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Cabel TV`],
    premium: false,
    price: 150,
    rating: 5,
    reviews: [
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 169,
        name: `John`,
        rating: 4,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-24T00:00:00.000Z`,
        reviewID: 183,
        name: `Victoria`,
        rating: 1,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-03-02T00:00:00.000Z`,
        reviewID: 147,
        name: `Victoria`,
        rating: 3,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nice, cozy, warm big bed apartment`,
    type: `apartment`,
  },
];
