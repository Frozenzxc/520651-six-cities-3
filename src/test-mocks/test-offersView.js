const OffersView = {
  ALL: `cities__`,
  NEARBY: `near-`,
};

export {OffersView};
