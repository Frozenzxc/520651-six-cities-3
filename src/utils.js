export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const parseOffer = (data) => {
  let offer = {};
  offer.bedrooms = data[`bedrooms`];
  offer.city = data[`city`];
  offer.description = data[`description`];
  offer.goods = data[`goods`];
  offer.host = {};
  offer.host.avatar = data[`host`][`avatar_url`];
  offer.host.id = data[`host`][`id`];
  offer.host.isPro = data[`host`][`is_pro`];
  offer.host.name = data[`host`][`name`];
  offer.id = data[`id`];
  offer.images = data[`images`];
  offer.isFavorite = data[`is_favorite`];
  offer.isPremium = data[`is_premium`];
  offer.location = data[`location`];
  offer.maxAdults = data[`max_adults`];
  offer.previewImage = data[`preview_image`];
  offer.price = data[`price`];
  offer.rating = data[`rating`];
  offer.title = data[`title`];
  offer.type = data[`type`];

  return offer;
};

export const parseReview = (data) => {
  let review = {};
  review.comment = data[`comment`];
  review.date = data[`date`];
  review.id = data[`id`];
  review.rating = data[`rating`];
  review.user = {};
  review.user.avatar = data[`user`][`avatar_url`];
  review.user.id = data[`user`][`id`];
  review.user.isPro = data[`user`][`is_pro`];
  review.user.name = data[`user`][`name`];

  return review;
};

export const getCitiesList = (offers) => {
  return [...new Set(offers.map((offer) => offer.city.name))];
};

export const noop = () => {};
