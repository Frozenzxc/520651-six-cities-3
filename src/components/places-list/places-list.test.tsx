import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import PlacesList from "./places-list";
import {AuthorizationStatus, OfferType} from "../../const";
import {offers} from "../../test-mocks/test-offers";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {BrowserRouter} from "react-router-dom";
import {noop} from "../../utils";

const mockStore = configureStore([]);

it(`PlacesList is rendered correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PlacesList
            addToFavorite={noop}
            authorizationStatus={AuthorizationStatus.AUTH}
            offers={offers}
            onCardHover={noop}
            onCardTitleClick={noop}
            offersView={OfferType.ALL}
          />
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`PlacesList is rendered correctly without offers`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PlacesList
            addToFavorite={noop}
            authorizationStatus={AuthorizationStatus.AUTH}
            offers={[]}
            onCardHover={noop}
            onCardTitleClick={noop}
            offersView={OfferType.ALL}
          />
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
