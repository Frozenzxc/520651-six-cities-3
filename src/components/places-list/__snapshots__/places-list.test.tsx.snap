// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList is rendered correctly 1`] = `
<div
  className="cities-places__list places__list tabs__content"
>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            160
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            192
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "84%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Waterfront with extraordinary view
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            350
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "88%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            196
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "66%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        >
          Loft Studio in the Central Area
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            116
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "84%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/5"
          onClick={[Function]}
        >
          Penthouse, 4-5 rooms + 5 balconies
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            892
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/6"
          onClick={[Function]}
        >
          Loft Studio in the Central Area
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="cities-places__card place-card"
    onMouseEnter={[Function]}
  >
    <div
      className="cities-image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            136
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/7"
          onClick={[Function]}
        >
          Waterfront with extraordinary view
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
</div>
`;

exports[`PlacesList is rendered correctly without offers 1`] = `
<div
  className="cities-places__list places__list tabs__content"
>
  No places to stay available
</div>
`;
